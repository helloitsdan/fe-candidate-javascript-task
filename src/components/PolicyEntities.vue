<template>
  <div>
    <h2>Pets</h2>

    <div v-if="!insured_entities">
      There are no associated pets.
    </div>

    <div v-if="insured_entities" class="columns is-multiline">
      <div
        v-for="entity in insured_entities"
        v-bind:key="entity.uuid"
        class="column is-one-third"
      >
        <div class="card background--white">
          <entity-information v-bind="entity" />
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import EntityInformation from "./EntityInformation.vue";

export default Vue.extend({
  name: "PolicyEntities",
  components: {
    EntityInformation,
  },
  props: {
    insured_entities: Array,
  }
});
</script>
