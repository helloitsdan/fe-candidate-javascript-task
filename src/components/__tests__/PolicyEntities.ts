import { shallowMount } from "@vue/test-utils";

import PolicyEntities from "../PolicyEntities.vue";

describe("PolicyEntities", () => {
  const fakePolicy = {
    entities: [],
    products: [],
  };

  it("renders", () => {
    const wrapper = shallowMount(PolicyEntities, {
      propsData: fakePolicy
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
