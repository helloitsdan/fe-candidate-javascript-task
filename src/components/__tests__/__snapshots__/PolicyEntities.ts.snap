// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyEntities renders 1`] = `
<div
  entities=""
>
  <h2>
    Pets
  </h2>
   
  <div>
    
    There are no associated pets.
  
  </div>
   
  <!---->
</div>
`;
